### PR TITLE
Pinch zoom features

### LINKED ISSUE #129: Expose a way to set scale/transform while preserving perspective/pan
Right now invoking `PinchZoom.setTransform({ scale })` uses a transform origin of `0,0`.  It might be nice to expose either a way to scale around a given center, or assume a default of the center of the `PinchZoom` region.

### PR DESCRIPTION
**Pull request by jakearchibald**
Do not merge, as this is not based on master, but please review!

Fixes #129.

### CODE REVIEW

File: src/components/Output/custom-els/PinchZoom/index.ts [hunk 1]
```diff
@@ -103,6 +117,59 @@ export default class PinchZoom extends HTMLElement {
...
+  scaleTo(scale: number, opts: ScaleToOpts = {}) {
+    let {
+      originX = 0,
+      originY = 0,
+    } = opts;
+
+    const {
+      relativeTo = 'content',
+      allowChangeEvent = false,
+    } = opts;
```
Comment (jakearchibald) [inline src/components/Output/custom-els/PinchZoom/index.ts]: Are these good defaults?

**Comment by jakearchibald** [comment]
I'd like to add the ability to animate between values (maybe an enhancement for browsers that support `el.animate`).

File: src/components/Output/index.tsx [hunk 2]
```diff
@@ -78,16 +77,16 @@ export default class Output extends Component<Props, State> {
 
   @bind
   zoomIn() {
-    this.setState({
-      scale: Math.min(this.state.scale * 1.25, 100),
-    });
+    if (!this.pinchZoomLeft) throw Error('Missing pinch-zoom element');
+
+    this.pinchZoomLeft.scaleTo(Math.min(this.state.scale * 1.25, 100), scaleToOpts);
```
Comment (developit) [inline src/components/Output/index.tsx]: Worth removing the min/max here to let pinch-zoom decide?

**Comment by developit** [comment]
LGTM. I do like the idea of animating using `.animate()` in browsers where it's available. Perhaps we could add an `animate?:bool` to the options?

File: src/components/Output/index.tsx (hunk 2 above)
Comment (jakearchibald) [reply src/components/Output/index.tsx]: Agreed. Will do that.

**Comment by jakearchibald** [comment]
@developit I've hard coded a min scale as it fixes the bug we have now. Making it configurable isn't completely straight forward, so I've made an issue for it https://github.com/GoogleChromeLabs/squoosh/issues/149.

**Comment by jakearchibald** [comment]
Shut up @googlebot

**Pull request merged**